### PR TITLE
(feat): add politician table to scheduler

### PR DESCRIPTION
**Pull request by RichardKruemmel**

### CODE REVIEW

File: src/service_layer/utils.py [hunk 1]
```diff
@@ -255,3 +236,39 @@ def prepare_poll_data(api_polls):
         for api_poll in api_polls
     ]
     return polls
+
+
+def prepare_politician_data(api_politicians):
```
Comment (takahiromitsui) [inline src/service_layer/utils.py:241]: The data format needs to be corrected.
I guess that you need to remove a pair of curly brackets.

File: src/service_layer/utils.py [hunk 2]
```diff
@@ -98,25 +98,6 @@ def fetch_missing_entities(self) -> List[models.Party]:
         return data_list
 
 
-def get_repository(
```
Comment (takahiromitsui) [inline src/service_layer/utils.py:101]: You should have added the politician's factory creation function to the map.

But I removed this function(get_repository) and moved it to the SqlAlchemyFactory class (=make it a method).

This refactoring enforces clearer modularity.
Pleas ensure that youll add entity and function pair to the map in the SqlAlchemyFactory.

File: src/service_layer/handlers.py [hunk 3]
```diff
@@ -39,10 +41,8 @@
 
 # Step 1
 def fetch_missing_entity(command: commands.FetchMissingEntity) -> List[Any]:
-    repo = utils.get_repository(
```
Comment (takahiromitsui) [inline src/service_layer/handlers.py:42]: I refactored SqlAlchemyFactory.

Please take a look at utils.py's comment below.

File: src/service_layer/handlers.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ def update_table(command: commands.UpdateTable):
         )
         return
     for entity, data in zip(command.entities, command.data):
-        repo = utils.get_repository(factory, entity)
```
Comment (takahiromitsui) [inline src/service_layer/handlers.py:81]: I refactored SqlAlchemyFactory.

Please take a look at utils.py's comment below.

File: src/api/repository.py [hunk 5]
```diff
@@ -151,48 +152,84 @@ def __init__(self, session):
...
 class SqlAlchemyFactory:
     """Factory class to create repositories"""
 
     def __init__(self, session):
         self.session = session
+        """Must add new repositories to the map to be able to create them"""
+        self.entity_factory_map = {
+            Entity.PARTY.value: self.create_party_repository,
+            Entity.PARTY_STYLE.value: self.create_party_style_repository,
+            Entity.POLITICIAN.value: self.create_politician_repository,
```
Comment (takahiromitsui) [inline src/api/repository.py:169]: I added the Politician entity and function pair.

Please ensure that you add it to the next one.

If it does not make sense, please look at the comment on utils.py(get_repository method).

**Comment by takahiromitsui** [comment]
After correcting the politician's data format, could you run a schedule task on your local machine to check if your changes work or not?

You'll see if it works on your logs or local database.

Just in case, I commented steps.

1. Make sure database connection URL (refer to src/db/connection.py)
2. Run redis (`redis-server` on your terminal)
3. Run the schedule task (`poetry run python -m src.entrypoints.redis_scheduled_tasks`)
4. Change schedule time on src/entrypoints/redis_scheduled_tasks.py
5. Check logs and database on your local


This is an example.
Screenshot 2024-04-26 at 10 20 23

**Review by takahiromitsui — CHANGES_REQUESTED** [review]
Please correct the data format and check if it works on your local machine.

File: src/service_layer/utils.py (hunk 1 above)
Comment (RichardKruemmel) [reply src/service_layer/utils.py:241]: Good spotting! I removed it and it works locally

**Review by takahiromitsui — APPROVED** [review]

**Pull request merged**